src/components/ToolbarButton.tsx

```diff
@@ -81,17 +83,40 @@ const ToolbarButton: FC<ToolbarButtonProps> = ({
   // TODO: type svg correctly
   const SVG = svg as React.FC<Icon>
 
+  // Get the direction if the shortcut is 'toggleSort'
+  const direction = useSelector((state: State) => {
+    if (shortcutId !== 'toggleSort') return null
+    return getCursorSortDirection(state)
+  })
+
+  /** Determines whether to animate the toolbar button based on the shortcut and state. */
+  const shouldAnimate = () => {
+    // Handle the 'toggleSort' shortcut separately
+    if (shortcutId === 'toggleSort') {
+      return direction === null || direction === 'Asc'
```

Comment:
The animations are functioning correctly (nice work!), but the reasoning is not obvious to me while reading the code.

@ohrytskov - Would you please explain why animating when there is "no direction" or "ascending" makes sense? I would think there should be an animation when changing to "descending" or "ascending", but I must be missing something.

Edit: Thinking about this more, I guess the reason is that we animate when changing _from_ these states. I.e., when changing from "no direction" or "ascending", we are changing _to_ "ascending" or "descending" and expect the animation. Is that accurate?

Reply:
Yes, that's accurate. We animate the state change when transitioning from "no direction" or "ascending" to "ascending" or "descending".